src/libcore/iter/traits/iterator.rs

```diff
@@ -1114,8 +1114,23 @@ pub trait Iterator {
     /// The `-3` is no longer there, because it was consumed in order to see if
     /// the iteration should stop, but wasn't placed back into the iterator.
     ///
+    /// Note that unlike [`take_while`] this iterator is **not** fused:
```

Comment:
I don't think this needs to be documented. The lack of a `FusedIterator` impl should be enough to know it's not fused. Although in general if you need a fused iterator you should just call `.fuse()` whether it happens to implement `FusedIterator` or not.

Additionally I think that the behavior after `.next()` has returned `None` should be left unspecified, unless of course there is some use case for that behavior. I'd imagine that if you need access to more elements after the predicate has returned `None` you'd be better off just using `.map(predicate)`.

Reply:
I disagree in that I think mentioning this iterator not being fused in the docs is a good thing. The lack of impls, especially for marker traits, is easily overlooked. 

However, I agree that we probably don't want to specify the behavior after the first `None` is returned. At least for now. If it turns out later that there are use cases for a well defined behavior, we can still add it. So I would replace the code example with one sentence and put it all in one paragraph. Something like this:

> Note that unlike `take_while` this iterator is **not** fused. It is also not specified what this iterator returns after the first `None` is returned.  If you need fused iterator, use `fuse`.